compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -102,6 +169,7 @@ impl<'a> Parser<'a> {
         allow_plus: AllowPlus,
         recover_qpath: RecoverQPath,
         allow_c_variadic: AllowCVariadic,
+        recover_return_sign: RecoverReturnSign,
```

Comment:
nit: can you re-order the arguments to this function so that the `allow_` arguments come before the `recover_` arguments?